src/%IPM/General/ModuleInfo.cls

```diff
@@ -20,6 +21,19 @@ Property VersionString As %String(%JSONFIELDNAME = "version", MAXLEN = 100, XMLN
 /// Is "" if the module was loaded using the load command (i.e. directly from a file and not from a repository).
 Property Repository As %String(MAXLEN = "", XMLPROJECTION = "NONE");
 
+/// Case-sensitive display name of the module as it appears in the module manifest
+Property DisplayNameInternal As %IPM.DataType.ModuleName;
+
+Property DisplayName As %IPM.DataType.ModuleName [ SqlComputeCode = { Set {*} = $Case({DisplayNameInternal}, "": {Name}, : {DisplayNameInternal})} ];
```

Comment:
Should have a JSON name of displayName probably?

Reply:
Nit: Should also be marked as calculated. I believe it shouldn't show up in the storage definition either then so may need to delete the entry for that and recompile persistent subclasses?